feat: [CDS-75242]: update documentation for module source on terrafor… (#2680)
harness/developer-hub: docs/continuous-delivery/cd-infrastructure/terraform-infra/run-a-terraform-plan-with-the-terraform-apply-step.md

The **Use Connector credentials** setting is limited to Harness Git Connectors using SSH authentication (not HTTPS) and a token.

> When configuring SSH Key for connector, exporting SSH Key with Passphrase for Module Source is not Supported. Configure a SSH Key without the passphrase.
> 
> Syntax examples to reference terraform module using ssh protocol:
> 
> ```json
> source = "git@github.com:your-username/your-private-module.git"
> or
> source = "git::ssh://git@github.com/your-username/your-private-module.git"
> ```

Reviewer: ```suggestion
When configuring the SSH key for the connector, exporting an SSH key with a passphrase for the module source is not supported. Configure an SSH Key without the passphrase.

Here are some syntax examples to reference the Terraform module using the SSH protocol:

```bash
source = "git@github.com:your-username/your-private-module.git"
```

or

```bash
source = "git::ssh://git@github.com/your-username/your-private-module.git"
```
```
